### ISSUE TITLE
[Bug][v1.12][Retail] Bot cannot find fishing bobber

### ISSUE THREAD

**Issue by Soromeister** [jsbots/AutoFish#24]
Using v1.12 on latest Retail (9.2.7 I think) 

I keep getting

```
[16:39:58] [WIN1] Casting fishing...

[16:40:00] [WIN1] Looking for the bobber...

[16:40:01] [WIN1] Can't find the bobber, recast.
```

Started WoW in DirectX 11 mode, Windowed (Fullscreen). This means I entered WoW once and changed it to DirectX 11, then I completely exit WoW and then start it again to make sure it starts in DirectX 11 mode.

Started WoW and did the below:
- Disabled all addons
- Removed all action bars except the main one which I cannot remove
- Zoomed in to FIrst Person mode
- Hid all the interface with ALT+Z

- Started the bot. Left default settings.
- Went to Dalaran Legion fountain. There is mostly blue there so I used the Red Threshold starting with a value of 30.
- That didn't work at all, so I increased the threshold by 1 and tried again.
- When reaching a threshold of 83, the bobber is cast, but then the bobber cannot be found and bot suggests that I lower the threshold.
- Lowering the threshold to 82 will not cast the bobber.
- Tried with literally every threshold value and the most I can get is to auto cast the bobber, but I can never get the bot to click on the bobber when it's time to do so.
- Switched to Blue also and it didn't work.
- Tried also in Zereth Mortis and without success.
- Tried to set a fishing zone by placing the rectangle window to indicate the area after which I hit the green checkmark to confirm it. Re-tried in Dalaran fountain as well as Zereth Mortis with ALL threshold values and the bobber cannot be found.

TLDR: Cannot find bobber after casting. I am using the "Fish" ability which is placed on the action button 1. There are no macros involved into this.vv

I am using Windows 11 22H2

![image](https://user-images.githubusercontent.com/45243695/196984509-4b77e678-b1d3-4586-b401-fe395f257fbb.png)
![image](https://user-images.githubusercontent.com/45243695/196984649-068d34b1-a812-4051-a981-9ac5a6deb505.png)
![image](https://user-images.githubusercontent.com/45243695/196984714-196f5b63-8eb4-44a1-977c-7cdf4c0e5e89.png)

**Comment by olesgeras** [comment]
Have you tried to increase cast animation delay? to 3000 / 3500 / 4000 / 4500

**Comment by Soromeister** [comment]
Forgot to mention it, but yes, I also tried that up to 15500 in increments of 500, as well as manually selecting "Custom window" and deselecting every other feature that would make it look like a human (logging out, random sleep, random reaction, like a human, etc).

I have 4k monitor (SDR) with 175% scale and NVidia Quadro 5000 if this matters for some reason.

I thought that the graphic settings might interfere, so I also tried with the absolute lowest graphic settings except for the resolution which is native 4k and in-game resolution of 50% which gives a 1080p rendered resolution.

**Comment by Soromeister** [comment]
Re-opened as I closed by misclick.

**Comment by Soromeister** [comment]
Made a [quick video recording](https://youtu.be/sw2V4tsdGrE) for better visibility.

**Comment by olesgeras** [comment]
Try to make your fishing zone smaller and make it cover only the water, this will allow you to decrease threshold even more because it won't include "reddish" environment. There's also an issue with some video drivers, the library the bot uses might not work with them properly, so sometimes it requires either update or downgrade. 

![image](https://user-images.githubusercontent.com/26541518/197119653-85fa5866-1bb7-4c2f-81a6-386d526d95a9.png)

**Comment by Bigbruver** [comment]
I just want to +1 this bug. Been trying to set up for two days and have followed all tips and tricks. The bot has not once been able to detect the bobber. As soon as the threshold is high enough (as low as 11 on pure blue water only) it's too high to detect the bright red feather in good weather.

**Comment by olesgeras** [comment]
> I just want to +1 this bug. Been trying to set up for two days and have followed all tips and tricks. The bot has not once been able to detect the bobber. As soon as the threshold is high enough (as low as 11 on pure blue water only) it's too high to detect the bright red feather in good weather.

Are you on win 11?

**Comment by olesgeras** [comment]
> > > I just want to +1 this bug. Been trying to set up for two days and have followed all tips and tricks. The bot has not once been able to detect the bobber. As soon as the threshold is high enough (as low as 11 on pure blue water only) it's too high to detect the bright red feather in good weather.
> > 
> > 
> > Are you on win 11?
> 
> I've run on both win 10 and win 11. Different systems with different video cards and driver versions (both nvidia) As well as both retail and wotlk classic versions. same bug everywhere

Well, that would mean the bot doesn't work at all and for everyone, but it works for the majority. We don't know whether it's a bug yet, so don't jump to conclusions. Record the screen of the whole fishing process as [Soromeister (https://github.com/Soromeister) did.

**Comment by Bigbruver** [comment]
https://clipchamp.com/watch/yrbjdSxXHYY

This is one example on classic with defaults left on. the threshold at 36 wouldn't cast, 41 couldn't find bobber. after video i found the exact threshold cutoff was 39 no cast, 40 no bobber finding.

**Comment by Bigbruver** [comment]
> > > > I just want to +1 this bug. Been trying to set up for two days and have followed all tips and tricks. The bot has not once been able to detect the bobber. As soon as the threshold is high enough (as low as 11 on pure blue water only) it's too high to detect the bright red feather in good weather.
> > > 
> > > 
> > > Are you on win 11?
> > 
> > 
> > I've run on both win 10 and win 11. Different systems with different video cards and driver versions (both nvidia) As well as both retail and wotlk classic versions. same bug everywhere
> 
> Well, that would mean the bot doesn't work at all and for everyone, but it works for the majority. We don't know whether it's a bug yet, so don't jump to conclusions. Record the screen of the whole fishing process as Soromeister did.

https://clipchamp.com/watch/yrbjdSxXHYY

This is one example on classic with defaults left on. the threshold at 36 wouldn't cast, 41 couldn't find bobber. after video i found the exact threshold cutoff was 39 no cast, 40 no bobber finding.

**Comment by olesgeras** [comment]
make a screenshot of your video settings in the game

**Comment by olesgeras** [comment]
Might have solved your issues with 1.13.0, test it.

**Comment by Soromeister** [comment]
I will close this ticket as I'll be looking for another bot that actually works. It's impossible to fill the entire screen with one colour only, so whatever detection is being used to detect the bobber, it doesn't do a great job at it.

The example I provided should work and the bot should detect the bobber model even though there are other elements in the game, such as the big ball in Zereth Mortis. Best it can do is niche areas as probably less than 10% of the entire game can be accommodated to fit the bot requirements so I'll just look for something else that I can simply cast and it works.

**Comment by jsbots** [comment]
>It's impossible to fill the entire screen with one colour only, so whatever detection is being used to detect the bobber, it doesn't do a great job at it.

The very first sentence tells that you haven't read the guide properly. What "one color only filling" you are talking about I don't know. The bot finds either the most red or the most blue cluster of points on the screen, which is enough for the _finding the bobber_ logic, because the feathers on the bobber correspond to such colors. You can also switch between blue and red colors to make it even more universal. 

>  the bot should detect the bobber model even though there are other elements in the game, such as the big ball in Zereth Mortis. 

This is the bot that requires a simple preliminary configuration, if you are unable to make it work with such a configuration then this isn't a bot for you indeed and it doesn't met your requirements of being "plug and play".

> Best it can do is niche areas as probably less than 10% of the entire game can be accommodated to fit the bot requirements

With proper repositioning and resizing of the **Fishing Zone** you can fish in any place in the game: hundreds of people use the bot in both DF and Classic and had no issues whatsoever. 

> I'll just look for something else

That's what you should have done in the first place without offending the project and thus the author. Good luck and Merry Christmas.
